vulkan_generator/struct_factories/generator.py

```diff
@@ -0,0 +1,451 @@
+                  {struct_factory_name(struct)}::~{struct_factory_name(struct)}() {{
+                  }}
+
+                  """)
+
+
+def generate_factory_setter_def(struct : str,
+                                member : str,
+                                member_data: types.VulkanStructMember,
+                                vulkan_metadata: types.VulkanMetadata) -> str :
+    return dedent(f"""
+                  void {struct_factory_name(struct)}::{set_member_name(member)}({set_member_arg_type(member_data, vulkan_metadata)} val) {{
+                      {member}_ = val;
+                  }}
+                  """)
+
+
+def generate_factory_getter_def(struct : str,
+                                member : str,
+                                member_data: types.VulkanStructMember,
+                                vulkan_metadata: types.VulkanMetadata) -> str :
+    return dedent(f"""
+                  {get_member_return_type(member_data, vulkan_metadata)} {struct_factory_name(struct)}::{get_member_name(member)}() const {{
+                      return {member}_;
+                  }}
+                  """)
+
+
+def generate_factory_size_def(struct : str) -> str :
+    return dedent(f"""
+                  size_t {struct_factory_name(struct)}::VkStructMemorySize() {{
+                      return sizeof({struct});
+                  }}
+                  """)
+
+
+def generate_factory_chain_size_def(struct : str, vulkan_metadata: types.VulkanMetadata) -> str :
+    if "pNext" in vulkan_metadata.types.structs[struct].members:
+        return dedent(f"""
+                      size_t {struct_factory_name(struct)}::PNextChainMemorySize() {{
+                          if(pNext_ != nullptr) {{
+                              size_t directNextSize = pNext_->VkStructMemorySize();
+                              if(directNextSize % sizeof(void*) != 0) {{
+                                  directNextSize += sizeof(void*) -(directNextSize % sizeof(void*));
+                              }}
+                              return directNextSize +pNext_->PNextChainMemorySize();
+                          }}
+                          return 0;
+                      }}
+                      """)
+    else:
+        return dedent(f"""
+                      size_t {struct_factory_name(struct)}::PNextChainMemorySize() {{
+                          return 0;
+                      }}
+                      """)
+
+
+def generate_factory_generate_def(struct : str) -> str :
+    return dedent(f"""
+                  {struct} {struct_factory_name(struct)}::Generate() {{
+                      {struct} ret;
+                      //TODO: Populate the fields of ret here.
+                      return ret;
+                  }}
+                  """)
+
+
+def generate_struct_factories_cpp(file_path: Path, vulkan_metadata: types.VulkanMetadata):
+    ''' Generates struct_factories.cc '''
+    with open(file_path, "w", encoding="ascii") as remapper_cpp:
```

Comment:
As a generic structure should we separate 

- What being generated
- How being generated
- How being written to a C++ file

Here we add stuff from hande remapper which should really not be here.